PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,476 @@
+  double nonPrefiringProbaMuon[7] = {
+      1., 1., 1., 1., 1., 1., 1.};  //0: central, 1: up, 2: down, 3: up stat, 4: down stat, 5: up syst, 6: down syst
+
+  for (const auto fluct : {fluctuations::central, fluctuations::up, fluctuations::down}) {
+    if (!missingInputEcal_) {
+      for (const auto& photon : thePhotons) {
+        double pt_gam = photon.pt();
+        double eta_gam = photon.eta();
+        if (pt_gam < 20.)
+          continue;
+        if (fabs(eta_gam) < 2.)
+          continue;
+        if (fabs(eta_gam) > 3.)
+          continue;
+        double prefiringprob_gam = getPrefiringRateEcal(eta_gam, pt_gam, h_prefmap_photon_, fluct);
+        nonPrefiringProbaPhoton[fluct] *= (1. - prefiringprob_gam);
+      }
+
+      //Now applying the prefiring maps to jets in the affected regions.
+      for (const auto& jet : theJets) {
+        double pt_jet = jet.pt();
+        double eta_jet = jet.eta();
+        double phi_jet = jet.phi();
+        if (pt_jet < 20.)
+          continue;
+        if (fabs(eta_jet) < 2.)
+          continue;
+        if (fabs(eta_jet) > 3.)
+          continue;
+        if (jetMaxMuonFraction_ > 0 && jet.muonEnergyFraction() > jetMaxMuonFraction_)
+          continue;
+        //Loop over photons to remove overlap
+        double nonprefiringprobfromoverlappingphotons = 1.;
+        bool foundOverlappingPhotons = false;
+        for (const auto& photon : thePhotons) {
+          double pt_gam = photon.pt();
+          double eta_gam = photon.eta();
+          double phi_gam = photon.phi();
+          if (pt_gam < 20.)
+            continue;
+          if (fabs(eta_gam) < 2.)
+            continue;
+          if (fabs(eta_gam) > 3.)
+            continue;
+          double dR2 = reco::deltaR2(eta_jet, phi_jet, eta_gam, phi_gam);
+          if (dR2 > 0.16)
+            continue;
+          double prefiringprob_gam = getPrefiringRateEcal(eta_gam, pt_gam, h_prefmap_photon_, fluct);
+          nonprefiringprobfromoverlappingphotons *= (1. - prefiringprob_gam);
+          foundOverlappingPhotons = true;
+        }
+        //useEMpt =true if one wants to use maps parametrized vs Jet EM pt instead of pt.
+        if (useEMpt_)
+          pt_jet *= (jet.neutralEmEnergyFraction() + jet.chargedEmEnergyFraction());
+        double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRateEcal(eta_jet, pt_jet, h_prefmap_jet_, fluct);
+
+        if (!foundOverlappingPhotons) {
+          nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet;
+        }
+        //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+        else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet) {
+          if (nonprefiringprobfromoverlappingphotons > 0.) {
+            nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
```

Comment:
@JanFSchulte  I think you need to add a line here to remove the prefiring weight associated to photons overlapping with this jet. 
`nonPrefiringProbaPhoton[fluct] *= 1/nonprefiringprobfromoverlappingphotons; `
This is what was done implicitly in the previous version of the code (where proba photon and proba jet were not factorized) 
Hopefully this is the source of the remaining difference in MINIAOD...

Reply:
I think my comment holds but is in fact not the source of the discrepancy which is likely related to an issue with the era modifier.